examples/WriteText/WriteText.ino

```diff
@@ -0,0 +1,123 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************
+ */
+
+ /**
+ ******************************************************************************
+* How to use this sketch
+*
+* This sketch uses the I2C interface for communicate with the NFC device.
+* It writes an NFC tag type Text and reads this same tag.
+* Check the pinout and the I2C interface used for communication.
+* Choose the content of the text by changing the content of text_write.
+*
+* We can see on the monitor window the message "Sytstem init done!" when the
+* NFC module is started and ready. Next, the tag is written, we wait few seconds,
+* we read the same tag and print it on the monitor window.
+*
+* You can test this application by connecting it with your smartphone.
+* If you want to read tag in RF mode (with smartphone) unplug the USB on the board.
+* The board should be power-off, or RF mode will not be available.
+*
+* You can write a Text tag with your smartphone.
+* On Android, donwload the NFC Tools.
+* Then start the app, check if NFC is activated on your smartphone.
+* Choose "ecrire" and then select the Text tag.
+* You can also read the text tag with the same app.
+ ******************************************************************************
+ */
+
+#include <Wire.h>
+#include <ArduinoSTL.h>
+#include <M24SR.h>
+
+#define SerialPort  Serial
+#define I2C2_SCL    PB10
+#define I2C2_SDA    PB11
+
+const uint8_t M24SR_ADDR=0xAC;
+const int     DEFAULT_GPO_PIN=PE4;
+const int     DEFAULT_RF_DISABLE_PIN=PE2;
+
+TwoWire dev_i2c(I2C2_SDA, I2C2_SCL);
+
+uint8_t done = 0;
+M24SR nfcTag(M24SR_ADDR, &dev_i2c, NULL, DEFAULT_GPO_PIN, DEFAULT_RF_DISABLE_PIN);
+
+/**
+ * Write a Ndef Txt message, wait and read the message from
+ * the Nfc tag.
+ *
+ */
+void setup() {
+  // Initialize serial for output.
+  SerialPort.begin(9600);
+
+  // Initialize I2C bus.
+  dev_i2c.begin();
+
+  nfcTag.begin(NULL);
+
+  SerialPort.println("System Init done: !");
+}
+
+void loop() {
+  const char *text_write = "hello world welcome on earth";     // Text to write in the tag
+  char text_read[255];                                         // Text read in the tag
```

Comment:
It can be good to initialize this value. (in case we see something in case of error during reading)